src/main/java/ch/njol/skript/lang/Variable.java

```diff
@@ -67,13 +65,14 @@ public class Variable<T> implements Expression<T>, KeyReceiverExpression<T>, Key
 	private final Class<? extends T>[] types;
 
 	private final boolean local;
+	private final boolean ephemeral;
```

Comment:
do you think it's worth it to have an enum for the scope of the variable (global, ephemeral, local) instead of multiple boolean fields? since they're mutually exclusive

Reply:
Yes, but I was about to make another pr with a more comprehensive rework of Variable lol

Reply:
I think it's within scope to do it in this pr, it's not really that much of a major change